src/main/java/nl/tudelft/semgroup4/GameEndedState.java

```diff
@@ -22,208 +20,406 @@
     public void init(GameContainer container, StateBasedGame game)
             throws SlickException {
         input = container.getInput();
-        
+
+        initializeTextUtilities();
+        initializeTextFields(container);
+        initializeTextFieldMouseOvers(container);
+        initializeContinueButton(container);
+
+    }
+
+    /**
+     * Setup for initializing extra data after the state has been made at the
+     * beginning of the app.
+     * 
+     * @param players
+     *            Players currently in the game
+     * @param won
+     *            True if the game has been won
+     */
+    public void setup(Player[] players, boolean won) {
+        this.won = won;
+        this.players = players;
+    }
+
+    /**
+     * Renders all elements on the screen.
+     * 
+     * @param container
+     *            container in which elements need to be rendered
+     * @param game
+     *            game in which elements need to be rendered
+     * @param graphics
+     *            graphics used to render the elements
+     */
+    @Override
+    public void render(GameContainer container, StateBasedGame game,
+            Graphics graphics) throws SlickException {
+
+        drawTitle(container);
+        drawColumnNames(container);
+        drawFirstPlayerInfo(container);
+
+        graphics.setColor(Color.white);
+        textFieldPlayer1.render(container, graphics);
+
+        drawRectangle(container, graphics);
+
+        if (players.length == 2) {
+            drawSecondPlayerInfo(container, graphics);
+        }
+
+        drawContinueButton(container);
+
+    }
+
+    /**
+     * Updates all elements on the screen.
+     * 
+     * @param container
+     *            container in which elements need to be updated
+     * @param game
+     *            game in which elements need to be updated
+     * @param delta
+     *            time between each frame
+     */
+    @Override
+    public void update(GameContainer container, StateBasedGame game, int delta)
+            throws SlickException {
+
+        // checks all mouse clicks in order to be able to select the text boxes
+        // and continue button.
+        if (input.isMousePressed(Input.MOUSE_LEFT_BUTTON)) {
+            if (mouseOverTextField1.isMouseOver()) {
+                textFieldPlayer1.setFocus(true);
+            } else if (mouseOverTextField2.isMouseOver()) {
+                textFieldPlayer2.setFocus(true);
+            } else if (mouseOverContinueButton.isMouseOver()) {
+                saveScore();
+                game.enterState(States.StartScreenState);
+            }
+        }
+
+    }
+
+    /**
+     * Returns the id of this state.
+     */
+    @Override
+    public int getID() {
+        return States.GameEndedState;
+    }
+
+    /**
+     * saves the score of the player(s) which have finished a game, the score is
+     * save with the corresponding filled out name.
+     */
+    public void saveScore() {
+        try {
```

Comment:
You should put this kind of logic in a controller for this state. 
Also watch the visibility of your methods, it only needs to be protected (when you put it in a controller).